### PR TITLE
Solutii test

### PR DESCRIPTION
**Pull request by marianrabagel**
Solutii + unit testing

### CODE REVIEW

File: 1/SumOfEvenNumbers.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            for (int i = 0; i < array.Length; i++)
+                if (isEven(array[i]))
+                    addToSum(array[i]);
+            return s;
+        }
+        private bool isEven(int number)
+        {
+            if(number % 2 == 0)
+                return true;
+            return false;
```
Comment (pirvudoru) [inline 1/SumOfEvenNumbers.cs]: How do you like this:

````c#
return number % 2 == 0;
````

File: 1/SumOfEvenNumbers.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            s = 0;
+        }
+
+        public int computeEvenNumbersSum(int[] array)
+        {
+            for (int i = 0; i < array.Length; i++)
+                if (isEven(array[i]))
+                    addToSum(array[i]);
+            return s;
```
Comment (pirvudoru) [inline 1/SumOfEvenNumbers.cs]: You are using an instance variable that gets modified when this method is executed. If you call this method more than once, you will get invalid results.

File: 1/SumOfEvenNumbers.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace AgileFreaks_test1
+{
+    public class SumOfEvenNumbers
+    {
+        private int s;
+
+        public SumOfEvenNumbers()
+        {
+            s = 0;
```
Comment (pirvudoru) [inline 1/SumOfEvenNumbers.cs]: If you indeed want that sum to be a member on an instance, make the array a hard dependency for the instance.

Comment (marianrabagel) [inline 1/SumOfEvenNumbers.cs]: If I would assign 0 value to my instance variable every time my method is
called it should resolve the problem of getting invalid results for
multiple calls.

File: 2/LargestTwoNumbers.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                checkMaxAndSecondMaxEquality();
+            }
+        }
+
+        private void checkMaxAndSecondMaxEquality()
+        {
+            if (areEqual())
+                secondMax = int.MinValue;
+        }
+        private bool areEqual()
```
Comment (pirvudoru) [inline 2/LargestTwoNumbers.cs]: The name of this method does not describe the implementation.

File: 2/LargestTwoNumbers.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        }
+        private bool isSecondMax(int number)
+        {
+            return (number < max) && (number > secondMax);
+        }
+        private bool isMax(int number)
+        {
+            return number > max;
+        }
+        private bool isNull(int[] array)
```
Comment (OvidiuBan) [inline 2/LargestTwoNumbers.cs]: Where is this method used?

File: 3/ArrayDuplicates.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace AgileFreaks_test1
+{
+    public class ArrayDuplicates
+    {
+        public bool hasDuplicates(int[] array)
+        {
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (array[array[i] % array.Length] > array.Length)
```
Comment (balauru) [inline 3/ArrayDuplicates.cs]: You are doing this calculation twice `array[i] % array.Length` you can extract in a variable

**Comment by balauru** [comment]
Mersi Marian, buna treaba, check your email

**Pull request closed without merging**